File: Data Structures & Algorithms/two-integer-sum/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {

        if(nums.length < 2) {
            return null;
        }

        const hashMap = {};

        for(let i = 0; i < nums.length; i++) {

            const difference = target - nums[i];

            if(hashMap[difference] === undefined) {
                hashMap[nums[i]] = i;
            } else {
                return [i, hashMap[difference]];
            }

        }

        return null;

    }
}

File: Data Structures & Algorithms/first-missing-positive/submission-3.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        
        hash_set = set(nums)

        target = 1

        for i in range(0, len(nums)):
            if target not in hash_set:
                return target
            else:
                target += 1

        return target